jgrapht-core/src/main/java/org/jgrapht/alg/matching/PathGrowingWeightedMatching.java

```diff
@@ -178,10 +164,10 @@ public double getMatchingWeight()
     }
 
     // the algorithm (no heuristics)
-    private Pair<Double, Set<E>> run(Graph<V, E> graph)
+    private Pair<Double, Set<E>> run()
```

Comment:
Can we eliminate the usage of Pair here and immediately return a DefaultMatching instead?